src/Execute/ExecutionEnv.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Command} from '../Backend/Command';
+
+interface ExecutionEnv {
+  name(): string;
+  envInfo(): string;
+  runCommand(command: Command): Command;
+}
+
+// This is a base ExeuctionEnv.
+// on New ExecutionEnv implements, need to add 3 things
+// 1. ExecutionEnv class itself
+// 2. ExecutionEnv Creation function : named `function getEnv(name: string): ExecutionEnv`
```

Comment:
1.  `create` would be better than `get` since it creates execution env
2. In our code, there are another `Env`. More specific term would be better, I think, .e.g., `Env` -> `ExecutionEnv`

https://github.com/Samsung/ONE-vscode/blob/d4f44b8eb382657b6a57574dbb3e6ad63df9d91e/src/Toolchain/ToolchainEnv.ts#L181

```suggestion
// 2. ExecutionEnv Creation function : named `function createExecutionEnv(name: string): ExecutionEnv`
```

Reply:
Oh I will fix it.